redis/db/redisbp/prometheus_test.go

```diff
@@ -0,0 +1,38 @@
+package redisbp
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type fakeClient redis.Client
+
+// PoolStats returns connection pool stats.
+func (c *fakeClient) PoolStats() *redis.PoolStats {
+	return &redis.PoolStats{
+		Hits:     1,
+		Misses:   2,
+		Timeouts: 3,
+
+		TotalConns: 4,
+		IdleConns:  5,
+		StaleConns: 6,
+	}
+}
+
+func TestRedisPoolExporter(t *testing.T) {
+	client := fakeClient{}
+
+	exporter := newExporter(
+		&client,
```

Comment:
nit: since only `*fakeClient` implements it, using the non-pointer as `client` is misleading, we should just do `client := &fakeClient{}` (or `client := new(fakeClient)`) on line 27 and just use `client` here, or just inline it here.

(also it's really unnecessary to make it a pointer receiver on line 14)

Reply:
Yes, I was going to do that, but forgot to fix it. :facepalm: